### DISCUSSION TITLE
Commenting implemented for non-advanced users

### DISCUSSION THREAD

**Discussion by SaturnTeamDev** [mediacms-io/mediacms#1048] [Show and tell]
On June 4th, 2024, [KyleMaas](https://github.com/KyleMaas) issued a fix which enabled registered (non-advanced) users to comment on videos in response to issue #216. The reply was linked to pull request #1023. However, a later code update must have broke his fix. His code crashed my single server site in a 502 error using Ubuntu 22.04. Now, I have amended his code and implemented commenting for regular users who can't upload. This is linked with implementing monetization for my site. Below, you'll find the updated file names and code:

### files/views.py (lines 25-33)

**from actions.models import USER_MEDIA_ACTIONS, MediaAction
from cms.custom_pagination import FastPaginationWithoutCount
from cms.permissions import IsAuthorizedToAdd, IsUserOrEditor, user_allowed_to_upload
from cms.permissions import IsAuthorizedToAddComment, IsUserOrEditor
from users.models import User**

### files/views.py (lines 1206-1208)

**permission_classes = (IsAuthorizedToAdd,)
permission_classes = (IsAuthorizedToAddComment,)
parser_classes = (JSONParser, MultiPartParser, FormParser, FileUploadParser)**

**Comment by mgogoulos** [comment]
Hi, I cannot understand what is happening here, what is added and what is removed. I also don't think the 502 error is related, based on the changes i see on the code. Keep in mind that docker/server install share the same code
